.github/workflows/test_changed_files.yaml

```
@@ -0,0 +1,29 @@
+on:
+  pull_request:
+    paths:
+      - 'changed-files/action.yml'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Get changed files
+        id: changed-files
+        uses: ./changed-files
+
+      - name: Check changed files output
+        run: |
+          changed_files="${{ steps.changed-files.outputs.changed-files }}"
+          expected_file="changed-files/action.yml"
+
+          if [[ $changed_files =~ $expected_file ]]; then
```

No need for regexes.
```suggestion
          if [ "$changed_files" = "$expected_file" ]; then
```

---

Wouldn't this break if `changed_files` includes any other file? For example this PR also includes `.github/workflows/test_changed_files.yaml` itself.

Perhaps `grep -F -q -x "$expected_file"`?

---

Ah, that's what you're after - yeah that would be better indeed.

Though I prefer long options for scripts:

```
echo $changed_files | grep --fixed-strings --quiet --line-regexp $expected_file
```